libraries/botbuilder-dialogs-adaptive/src/skills/adaptiveSkillDialog.ts

```diff
@@ -0,0 +1,137 @@
+/**
+ * @module botbuilder-dialogs-adaptive
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { SkillDialog, SkillDialogOptions, DialogContext, DialogTurnResult, DialogManager, BeginSkillDialogOptions } from 'botbuilder-dialogs';
+import { BoolExpression, StringExpression } from 'adaptive-expressions';
+import { TemplateInterface } from '../template';
+import { Activity, ActivityTypes } from 'botbuilder-core';
+
+const GLOBAL_SKILL_OPTIONS = Symbol('globalSkillOptions');
+
+export class AdaptiveSkillDialog extends SkillDialog {
+
+    /**
+     * Optional expression which if is true will disable this action.
+     */
+    public disabled?: BoolExpression;
+
+    /**
+     * Value indicating whether the new dialog should process the activity.
+     * 
+     * @remarks
+     * The default for this will be true, which means the new dialog should not look at the activity. 
+     * You can set this to false to dispatch the activity to the new dialog.
+     */
+    public activityProcessed = new BoolExpression(true);
+
+    /**
+     * Optional property path to store the dialog result in.
+     */
+    public resultProperty?: StringExpression;
+
+    /**
+     * The Microsoft App ID that will be calling the skill.
+     * 
+     * @remarks
+     * Defauls to a value of `=settings.MicrosoftAppId` which retrievs the bots ID from settings.
+     */
+    public botId = new StringExpression('=settings.MicrosoftAppId');
+
+    /**
+     * The callback Url for the skill host.
+     * 
+     * @remarks
+     * Defauls to a value of `=settings.SkillHostEndpoint` which retrieves the endpoint from settings.
+     */
+    public skillHostEndpoint = new StringExpression('=settings.SkillHostEndpoint');
+
+    /**
+     * The Microsoft App ID for the skill.
+     */
+    public skillAppId: StringExpression;
+
+    /**
+     * The `/api/messages` endpoint for the skill.
+     */
+    public skillEndpoint: StringExpression;
+
+    /**
+     * Template for the activity.
+     */
+    public activity: TemplateInterface<Partial<Activity>>;
+
+    /**
+     * Optional. The OAuth Connection Name for the Parent Bot.
+     */
+    public connectionName: StringExpression;
+
+    /**
+     * Creates a new `AdaptiveSkillDialog instance.
+     * @param options Optional options used to configure the skill dialog.
+     */
+    constructor(options?: SkillDialogOptions) {
+        super(Object.assign({ skill: {} } as SkillDialogOptions, options));
+    }
+
+    public async beginDialog(dc: DialogContext, options: Partial<BeginSkillDialogOptions>): Promise<DialogTurnResult> {
```

Comment:
@stevenic I think `options` should still be optional, especially since we're not using it.

[`SkillDialog.beginDialog(dc: DialogContext, options?: any)`](https://github.com/microsoft/botbuilder-js/blob/f443f0be3521b2b9494e91055a50ef8ddaa231cd/libraries/botbuilder-dialogs/src/skillDialog.ts#L61)

Should it just be removed? Or do we keep it and add a comment?

Reply:
`options` should be optional here

Reply:
And we should potentially change it back to `any`; the base class will validate the structure of the parameter